sycl/include/sycl/ext/oneapi/experimental/uniform.hpp

```diff
@@ -0,0 +1,117 @@
+//==------ uniform.hpp - SYCL uniform extension --------*- C++ -*-----------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// ===--------------------------------------------------------------------=== //
+// Implemenation of the sycl_ext_oneapi_uniform extension.
+// https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_uniform.asciidoc
+// ===--------------------------------------------------------------------=== //
+
+#pragma once
+
+// SYCL extension macro definition as required by the SYCL specification.
+// 1 - Initial extension version. Base features are supported.
+#define SYCL_EXT_ONEAPI_UNIFORM 1
+
+#include <type_traits>
+
+// Forward declarations of types not allowed to be wrapped in uniform:
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+struct sub_group;
```

Comment:
I still don't understand why this has to be the `ext::oneapi` version of `sub_group`.  `sycl::sub_group` should exist (see https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/sub_group.hpp).  Is this required because the definition of the class is currently in `sycl::ext::oneapi::`, and the version in `sycl::` is only an alias?

Reply:
Yes, sycl::sub_group is an alias for oneapi::sub_group currently - see https://github.com/intel/llvm/blob/433a073e0b740451f92126f1b3c6075784ee2a93/sycl/include/CL/sycl/sub_group.hpp#L16. I need to repeat the aliasing, otherwise the forward declaration leads to later conflicts with actual definition. But with this aliasing, the `invoke_simd` API uses `sycl::sub_group` argument.

Does this address the concern?

Reply:
Yes, thanks for the explanation.